### PR TITLE
M2-4827: Post password reset redirection

### PR DESCRIPTION
**Pull request by sultanofcardio**
### 📝 Description

🔗 [Jira Ticket M2-4827](https://mindlogger.atlassian.net/browse/M2-4827)

I've redirected the user to the login page after successful password redirection and displayed a success banner on that page. I've also removed the `backToLogin` button, and the form success message.

### 📸 Screenshots

https://github.com/ChildMindInstitute/mindlogger-web-refactor/assets/14842108/ed8c47e1-7cb5-46b9-b415-4bc940dafb58

### 🪤 Peer Testing
Example test step:

- Start up the API and the webapp
- Navigate to the web app landing page
- Click the forgot password link
- Enter your email address and click *Get Reset It*
- Click on the link in your email (Go to http://localhost:8025/ for local testing)
- Enter the new password
  **Expected outcome:** You should be redirected to the login screen and see a banner displayed for 5 seconds

### ✏️ Notes

I'm using `useRef` as an escape hatch in the `LoginPage` component to prevent the banner from showing twice in strict mode. This is arguably a hack and should be generalised, but I don't know if it should be part of this ticket. I'd welcome some feedback on that

### CODE REVIEW

**Review by farmerpaul — APPROVED** [review]
Looks and works great! Just provided some feedback, but I think this is good to merge without any changes.

File: src/pages/Login/index.tsx [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
 import { Mixpanel } from "~/shared/utils"
 
 const DownloadMobileLinks = lazy(() => import("~/widgets/DownloadMobileLinks"))
 
 function LoginPage() {
   const { t } = useLoginTranslation()
   const location = useLocation()
+  const navigate = useNavigate()
+  const { showSuccessNotification } = useNotification()
+  const notificationRef = useRef<Notification | null>(null)
```
[the remark this answers is not in the document]
Comment (sultanofcardio) [reply src/pages/Login/index.tsx:20]: Agreed. Thanks for the feedback!

File: src/pages/Login/index.tsx [hunk 2]
```diff
@@ -1,25 +1,46 @@
...
+      navigate(window.location.pathname, {
+        state: {
+          ...location.state,
+          isPasswordReset: undefined,
+        },
+        replace: true,
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
```
[the remark this answers is not in the document]
Comment (sultanofcardio) [reply src/pages/Login/index.tsx:44]: Nah, I think that's a good idea. I'll implement it

**Review by moiskillnadne — APPROVED** [review]
Looks good to me

**Pull request merged**